src/core/qgsstoredexpressionmanager.h

```diff
@@ -0,0 +1,158 @@
+/***************************************************************************
+                          qgsstoredexpressionmanager.h
+                             -------------------
+    begin                : August 2019
+    copyright            : (C) 2019 David Signer
+    email                : david at opengis dot ch
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSSTOREDEXPRESSIONMANAGER_H
+#define QGSSTOREDEXPRESSIONMANAGER_H
+
+#include "qgis_core.h"
+#include <QString>
+#include <QObject>
+#include <QUuid>
+
+
+#ifdef SIP_RUN
+% ModuleHeaderCode
+#include <qgsstoredexpressionmanager.h>
+% End
+#endif
+
+class QDomNode;
+class QDomDocument;
+
+/**
+ * \ingroup core
+ * Stored expression containing name, content (expression text) and a category tag.
+ * \since QGIS 3.10
+ */
+struct CORE_EXPORT QgsStoredExpression
+{
+
+  /**
+   * Categories of use cases
+   * FilterExpression for stored expressions to filter attribute table
+   */
+  enum Category
+  {
+    FilterExpression = 1 << 0 //!< Expressions to filter features
+  };
+
+#ifndef SIP_RUN
+  QgsStoredExpression() = default;
+
+  QgsStoredExpression( QString name, QString expression, Category tag = Category::FilterExpression )
+    : id( QUuid::createUuid().toString() ),
+      name( name ),
+      expression( expression ),
+      tag( tag )
+  {}
+#endif
+
+  QString id;
+  QString name;
+  QString expression;
+  Category tag;
+};
+
+/**
+ * \ingroup core
+ * Manages stored expressions regarding creation, modification and storing in the project
+ * \since QGIS 3.10
+ */
+class CORE_EXPORT QgsStoredExpressionManager : public QObject
+{
+    Q_OBJECT
```

Comment:
No need for this to be a QObject - there's no signals or slots

Reply:
So you suggest not to use QObject and not to create a pointer from it in `QgsVectorLayer` and just return the reference, right?
That would be fine for me. We do not reassign it and we use only one instance.

@m-kuhn you defended my original approach using a pointer above. Are there any big reasons for using it versus the reference when it's not a `QObject` anymore?

Reply:
I think we still never want a copy of it (and maybe eventually promote it to a QObject and have it send a signal when new expressions are added etc.)
So in Qt terms "it's more an identity than a value".

Reply:
I'm fine with it being a qobject or pointer. We just need to be sure if it IS a qobject then it's parented to the layer. And also be aware that a qobject without signals/slots can throw warnings.

Reply:
Okay, thanks.